Multisig/UI/WalletConnectV2/WalletConnectManager.swift

```diff
@@ -183,15 +199,98 @@ class WalletConnectManager {
-        Sign.instance.getSessions().filter({topics.contains($0.topic)})
+        Sign.instance.getSessions().filter({ topics.contains($0.topic) })
     }
+
+    private func handle(request: Request) {
+        dispatchPrecondition(condition: .notOnQueue(.main))
+
+        guard let session = getSessions(topics: [request.topic]).first else {
+            reject(request: request)
+            return
+        }
+
+
+        guard let safe = Safe.by(topic: request.topic) else {
+            reject(request: request)
+            return
+        }
+
+        if request.method == "eth_sendTransaction" {
+
+            // make transformation of incoming request into internal data types
+            // and fetch information about safe from the request
+
+            guard let safeInfo = try? App.shared.clientGatewayService.syncSafeInfo(
+                    safeAddress: safe.addressValue, chainId: safe.chain!.id!)
+            else {
+                reject(request: request)
+                return
+            }
+
+            safe.update(from: safeInfo)
+
+            guard let ethereumTransaction = try? request.params.get([EthereumTransaction].self).first,
+                  let transaction = Transaction(transaction: ethereumTransaction, safe: safe)
+            else {
+                reject(request: request)
+                return
+            }
+
+            guard !safe.isReadOnly else {
+                DispatchQueue.main.async {
+                    App.shared.snackbar.show(message: "Please import Safe Owner Key to initiate WalletConnect transactions")
+                }
+                reject(request: request)
+                return
+            }
+
+            DispatchQueue.main.async { [unowned self] in
+                // present confirmation controller
+
+                let confirmationController = WCIncomingTransactionRequestViewController(
+                        transaction: transaction,
+                        safe: safe,
+                        dAppName: session.peer.name,
+                        dAppIconURL: URL(string: session.peer.icons.first ?? ""))
+
+                confirmationController.onReject = { [unowned self] in
+
+                    reject(request: request)
+                }
+
+                confirmationController.onSubmit = { nonce, safeTxHash in
+                    self.sign(request: request, response: AnyCodable(safeTxHash))
+                }
+
+                let sceneDelegate = UIApplication.shared.connectedScenes.first!.delegate as! SceneDelegate
+                sceneDelegate.present(ViewControllerFactory.modal(viewController: confirmationController))
+            }
+        } else {
+            DispatchQueue.global(qos: .background).async {
+                do {
+                    let rpcURL = safe.chain!.authenticatedRpcUrl
+                    let result = try AnyCodable(any: App.shared.nodeService.rawCall(payload: request.asJSONEncodedString(),
```

Comment:
This is the same behaviour as the WC1 code. It doesn't make much sense for  `eth_signTransaction`, `personal_sign`. `eth_sign` (standard) and `eth_signTypedData`,  but these are currently impossible because the contract doesn't have the key for the address it is deployed to. This might change for `eth_sig`/`eth_singTypedData` with the advent of 1271.